### PR TITLE
Implement nested markdown lists

### LINKED ISSUE #16003: [Markdown] Nested bulleted lists don't work in Julia Markdown
I have some function documentation that looks like this:

```
* ABC
   * ABC-1
   * ABC-2
   * ABC-3
* DEF
* GHI
```

I expect it to render like this:
- ABC
  - ABC-1
  - ABC-2
  - ABC-3
- DEF
- GHI

But it actually renders like this:
- ABC
- ABC-1
- ABC-2
- ABC-3
- DEF
- GHI

### PR DESCRIPTION
**Pull request by MichaelHatherly**
Adds basic support for writing nested lists, as well as any other "block"-level markdown element such as code blocks, tables, quote blocks, etc. within a single list item.

Since lists can now contain block elements some additional spacing has been added between items when output in the terminal to make it easier to differentiate between elements.

Fixes #16003.

---

`genstdlib` changes appear to all be simple whitespace changes that don't affect the final docs. ~~(Only exception I've found is the `IOContext` change where an indented code block is lost due to a list appearing before it. This can just be replaced with a fenced code block though, so not a problem really.)~~

~~Still todo: enable starting ordered lists from numbers other than `1`, which just needs several changes throughout the `markdown/render` folder.~~

### CODE REVIEW

**Comment by MichaelHatherly** [comment]
And #11249 could be closed with this change:

```diff
diff --git a/base/markdown/Common/block.jl b/base/markdown/Common/block.jl
index da53983..386207d 100644
--- a/base/markdown/Common/block.jl
+++ b/base/markdown/Common/block.jl
@@ -226,6 +226,7 @@ List(xs...) = List(vcat(xs...))
 const BULLETS = r"^ {0,3}(\*|\+|•|-)( |$)"
 const NUM_OR_BULLETS = r"^ {0,3}(\*|•|\+|-|\d+(\.|\)))( |$)"
 
+@breaking true ->
 function list(stream::IO, block::MD)
     withstream(stream) do
         bullet = startswith(stream, NUM_OR_BULLETS; eat = false)
```

If there's any support for that then I'm happy to add it as well.

File: doc/stdlib/io-network.rst [hunk 1]
```diff
@@ -415,9 +415,7 @@ General I/O
    * use ``(key => value) in dict`` to see if this particular combination is in the properties set
    * use ``get(dict, key, default)`` to retrieve the most recent value for a particular key
 
-   .. code-block:: julia
-
-       IOContext(io::IO, context::IOContext)
+     IOContext(io::IO, context::IOContext)
```
Comment (tkelman) [inline doc/stdlib/io-network.rst]: maybe these should be separate docstrings?

File: doc/stdlib/parallel.rst [hunk 2]
```diff
@@ -173,29 +173,19 @@ General Parallel Computing Support
...
-
-       sshflags=`-i /home/foo/bar.pem`
-
-   * ``max_parallel``\ : specifies the maximum number of workers connected to in parallel at a host.                   Defaults to 10.
-
-   * ``dir``\ : specifies the working directory on the workers. Defaults to the host's current          directory (as found by ``pwd()``\ )
-
-   * ``exename``\ : name of the ``julia`` executable. Defaults to ``"$JULIA_HOME/julia"`` or              ``"$JULIA_HOME/julia-debug"`` as the case may be.
-
+     sshflags=``-i /home/foo/bar.pem``
```
Comment (tkelman) [inline doc/stdlib/parallel.rst]: don't think this is right, should be keeping the code for the whole line and the single-backticks as julia code if possible

File: doc/stdlib/io-network.rst (hunk 1 above)
Comment (MichaelHatherly) [inline doc/stdlib/io-network.rst]: Yes, was wondering whether that would be better perhaps. Should I split them in this PR or just fix up the code block for now?

File: doc/stdlib/parallel.rst (hunk 2 above)
Comment (MichaelHatherly) [inline doc/stdlib/parallel.rst]: Same problem as with `IOContext`, using a fenced code block should fix this too.

File: doc/stdlib/io-network.rst (hunk 1 above)
Comment (tkelman) [inline doc/stdlib/io-network.rst]: either would work, separating the docstrings could technically be a separate PR to be pedantic

**Comment by MichaelHatherly** [comment]
Rebased, should be good to go now? This is probably the last markdown feature we needed to make the RST -> MD manual conversion as lossless as possible I think.

**Comment by tkelman** [comment]
lgtm

**Comment by MichaelHatherly** [comment]
win32 error looks unrelated, will merge shortly as is.
